osbrain/tests/test_nameserver.py

```diff
@@ -77,6 +96,28 @@ def test_nameserver_proxy_shutdown_with_agents():
     ns.shutdown()
 
 
+def test_nameserver_shutdown_with_agents():
+    """
+    Shutdown a name server with agents that wait some time before shutting
+    down.
+    """
+    nameserver, addr = get_nameserver_and_address()
+
+    class NewAgent(Agent):
+        def __init__(self, *args, **kwargs):
+            super(NewAgent, self).__init__(*args, **kwargs)
+
+        @Pyro4.oneway
+        def shutdown(self):
+            time.sleep(0.3)
+            super().shutdown()
+
+    run_agent('Agent0', nsaddr=addr, base=NewAgent)
+    run_agent('Agent1', nsaddr=addr, base=NewAgent)
```

Comment:
Is this the reason why you implemented `get_nameserver_and_address()` above? If so, can't you just omit the `nsaddr` parameter? (remember to pass the `nsproxy` fixture as parameter to the `test_nameserver_shutdown_with_agents()` function).

Also, if you really needed to get the address of the name server, I think you can do `nsproxy.addr()`. :wink: